ceph-dev-new-setup/build/build

```diff
@@ -64,6 +64,28 @@ else
     echo "forcing."
 fi
 
+# This is a dev release, enable some debug cmake configs. Note: it has been
+# this way since at least 35e1a715. It's difficult to tell when or even if ceph
+# was ever properly built with debugging configurations for QA as there are
+# corresponding changes in ceph with the switch to cmake which makes this
+# challenging to evaluate.
+#
+# It's likely that it was wrongly assumed that cmake would set the build type
+# to Debug because the ".git" directory would be present. This is not the case
+# because the "make-dist" script (executed below) creates a git tarball that is
+# used for the actual untar/build. See also:
+#
+# https://github.com/ceph/ceph/pull/53800
+#
+# Addendum and possibly temporary restriction: only enable these for branches
+# ending in "debug". The
+if [[ "$BRANCH" == *debug ]]; then
+  CEPH_EXTRA_CMAKE_ARGS+=" -DCMAKE_BUILD_TYPE=Debug -DWITH_CEPH_DEBUG_MUTEX=ON"
```

Comment:
Just noting that `WITH_CEPH_DEBUG_MUTEX=ON` should be implied in `CMAKE_BUILD_TYPE=Debug`.  We rely on that in a typical vstart and other cases:

https://github.com/ceph/ceph/blob/0deb52e4242c7c2c0ba2181f5759930d66637618/CMakeLists.txt#L361-L362

Reply:
That's right. I think it's good to set it here too as that implied behavior may change in the future accidentally or otherwise.